packages/eip-7702-internal-rpc-middleware/src/methods/wallet_getAccountUpgradeStatus.test.ts

```diff
@@ -0,0 +1,47 @@
+import type {
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+  Json,
+} from '@metamask/utils';
+
+import { walletGetAccountUpgradeStatus } from './wallet_getAccountUpgradeStatus';
+import { getAccountUpgradeStatus } from '../hooks/getAccountUpgradeStatus';
+import type { GetAccountUpgradeStatusHooks } from '../types';
+
+jest.mock('../hooks/getAccountUpgradeStatus');
+
+describe('walletGetAccountUpgradeStatus', () => {
+  it('calls getAccountUpgradeStatus hook with correct parameters and sets result', async () => {
+    const req = {} as JsonRpcRequest<Json[]> & { origin: string };
```

Comment:
Can we make real requests and responses rather than partial ones? Otherwise we're not really testing that the types are correct, we're just assuming that they are.